### PR TITLE
feat: allow explicit file extension field

### PR DESCRIPTION
**Pull request by yifanwu-kolena**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

**Review by ly-kolena — APPROVED** [review]

**Review by munkyshi — COMMENTED** [review]
Is there any risk of someone interpreting this as `file_type` of `image` (and putting in data type)? Should we handle that?

**Comment by yifanwu-kolena** [comment]
verified FE can load locator without file extension properly
Screen Shot 2024-06-17 at 2 28 42 PM

File: docs/dataset/advanced-usage/formatting-your-datasets.md [hunk 1]
```diff
@@ -50,10 +50,14 @@ Kolena will look for the following fields when displaying datapoints:
...
 | `text`     | Raw text input for text based models.                                                                                                       |
+| `file_type`| Optional field indicating the type of file (e.g., `jpg`, `mp4`). Used to assist in the inference of the datatype.                           |
 
-A locator needs to have correct extensions for the corresponding file type. For example an image should be in a format
+The `locator` needs to have correct extensions for the corresponding file type. For example an image should be in a format
 such as `.jpg` or `.png`, whereas locators for audio data should be in forms like `.mp3` or `.wav`.
 
+The optional `file_type` field can be used to explicitly provide the file type information
+when extensions cannot be included in the `locator`.
+
```
Comment (yifanwu-kolena) [inline docs/dataset/advanced-usage/formatting-your-datasets.md]: > Is there any risk of someone interpreting this as file_type of image (and putting in data type)? Should we handle that?

@munkyshi, i don't think we want to handle it for now, but just updated the docs to address the concern. does this make sense to you?

Screen Shot 2024-06-17 at 2 42 13 PM

Comment (munkyshi) [reply docs/dataset/advanced-usage/formatting-your-datasets.md]: Should we call it `file_extension` in that case?

Comment (yifanwu-kolena) [reply docs/dataset/advanced-usage/formatting-your-datasets.md]: yeah, `Explicit is better than implicit.`.
just rename it in [this commit](https://github.com/kolenaIO/kolena/pull/620/commits/207b30cebd9d5a0451abd99308e95fa69644dae1)

**Review by munkyshi — APPROVED** [review]

**Pull request merged**